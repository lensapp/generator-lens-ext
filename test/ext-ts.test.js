/* eslint-env jest */

const path = require("path");
const fs = require("fs");

const assert = require("yeoman-assert");
const { run } = require("yeoman-test");

test("ext-ts generator works as expected", async () => {
  const name = "my-test-ext";
  const resultPath = await run(path.join(__dirname, "../generators/app")).withPrompts({
    type: "ext-ts",
    name,
    description: "randome description",
    publisher: "random publisher",
    gitInit: true,
    pkgManager: "yarn",
    installDependencies: false
  });
  console.log("resultPath", resultPath);
  const files = [
    "README.md", "package.json", "webpack.config.js", "tsconfig.json", ".gitignore",
    "main.ts", "renderer.tsx", "page.tsx",
    ".eslintrc.js", ".eslintrc.main.js", ".eslintrc.renderer.js"
  ].map((fileName) => `${resultPath}/${name}/${fileName}`);
  assert.file(files);
  const packageJSON = JSON.parse(fs.readFileSync(`${resultPath}/${name}/package.json`, "utf8"));
  const tsconfig = JSON.parse(fs.readFileSync(`${resultPath}/${name}/tsconfig.json`, "utf8"));
  expect(packageJSON).toMatchSnapshot();
  expect(tsconfig).toMatchSnapshot();
});
