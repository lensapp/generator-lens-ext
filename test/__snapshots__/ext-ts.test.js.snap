// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ext-ts generator works as expected 1`] = `
Object {
  "contributes": Object {
    "cloudProviders": Array [],
    "crds": Array [],
    "kubernetesDistros": Array [],
  },
  "description": "randome description",
  "devDependencies": Object {
    "@k8slens/extensions": "^4.0.0-alpha.5",
    "@types/node": "^12.12.9",
    "@types/react": "^16.9.56",
    "eslint": "^7.13.0",
    "ts-loader": "^8.0.11",
    "typescript": "^4.0.5",
    "webpack": "^4.44.2",
    "webpack-cli": "^3.3.11",
  },
  "engines": Object {
    "node": ">=12.0 <13.0",
  },
  "keywords": Array [
    "lens",
    "extension",
    "k8slens",
    "kubernetes",
  ],
  "main": "dist/main.js",
  "name": "my-test-ext",
  "publisher": "random publisher",
  "renderer": "dist/renderer.js",
  "scripts": Object {
    "build": "npm run clean && webpack",
    "clean": "rm -rf ./dist",
    "start": "webpack --watch",
  },
  "version": "0.0.1",
}
`;

exports[`ext-ts generator works as expected 2`] = `
Object {
  "compilerOptions": Object {
    "allowSyntheticDefaultImports": true,
    "declaration": false,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "jsx": "react",
    "lib": Array [
      "ESNext",
      "DOM",
      "DOM.Iterable",
    ],
    "module": "CommonJS",
    "moduleResolution": "Node",
    "noImplicitAny": true,
    "outDir": "dist",
    "skipLibCheck": true,
    "sourceMap": false,
    "strict": false,
    "target": "ES2017",
  },
  "exclude": Array [
    "node_modules",
    "*.js",
  ],
  "include": Array [
    "./*.ts",
    "./*.tsx",
  ],
}
`;
